tests/test_with_credential.rs

```diff
@@ -1,166 +1,177 @@
-use rspotify::client::Spotify;
-use rspotify::oauth2::SpotifyClientCredentials;
-use rspotify::senum::{AlbumType, Country};
+mod common;
 
-use std::sync::Mutex;
+use common::maybe_async_test;
+use rspotify::client::{Spotify, SpotifyBuilder};
+use rspotify::oauth2::CredentialsBuilder;
+use rspotify::senum::{AlbumType, Country};
 
-use lazy_static::lazy_static;
+use maybe_async::maybe_async;
 
-lazy_static! {
-    // Set client_id and client_secret in .env file or
-    // export RSPOTIFY_CLIENT_ID="your client_id"
-    // export RSPOTIFY_CLIENT_SECRET="secret"
-    static ref CLIENT_CREDENTIAL: Mutex<SpotifyClientCredentials>
-        = Mutex::new(SpotifyClientCredentials::default().build());
-}
+/// Generating a new basic client for the requests.
+#[maybe_async]
+pub async fn creds_client() -> Spotify {
```

Comment:
No. I initially wanted to reuse the same client, but ran into lots of different issues related to concurrency or `lazy_static` (not sure if you remember when I was using `async_once`, but this had race conditions for some reason). I then undid my changes to use different clients, because tests should be ran in an "isolated" environment anyway.